trpcultivate_phenotypes/src/Plugin/Validators/ValidDelimitedFile.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/**
+ * @file
+ * Contains validator plugin definition.
+ */
+
+namespace Drupal\trpcultivate_phenotypes\Plugin\Validators;
+
+use Drupal\trpcultivate_phenotypes\TripalCultivateValidator\TripalCultivatePhenotypesValidatorBase;
+use Drupal\trpcultivate_phenotypes\TripalCultivateValidator\ValidatorTraits\FileTypes;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Validate a line in a data file is properly delimited.
+ *
+ * @TripalCultivatePhenotypesValidator(
+ *   id = "valid_delimited_file",
+ *   validator_name = @Translation("Valid Delimited File Validator"),
+ *   input_types = {"raw-row"}
+ * )
+ */
+class ValidDelimitedFile extends TripalCultivatePhenotypesValidatorBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * This validator requires the following validator traits:
+   * - FileTypes - getFileMimeType: get the MIME type of the input file.
+   */
+  use FileTypes;
+
+  /**
+   * The key used by the setter method to create a validator configuration element
+   * in the context array, as well as the key used by the getter method
+   * to reference and retrieve the element values.
+   *
+   * @var string
+   */
+  private $validator_context_key = 'ValidDelimitedFile';
+
+
+  /**
+   * Constructor.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition){
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+    );
+  }
+
+  /**
+   * Perform validation of a data file raw row.
+   * Checks include:
+   *  - Line is not empty.
+   *  - It has some delimiter used to separate values.
+   *  - When split, the number of values returned is equal to the expected number of values
+   *    set by the validator setter method.
+   *
+   * @param string $raw_row
+   *   A line in the data file that can be the headers row (line no. 1) or a data row.
+   *
+   * @return array
+   *   An associative array with the following keys.
+   *     - case: a developer focused string describing the case checked.
+   *     - valid: either TRUE or FALSE depending on if the raw row is delimited or not.
+   *     - failedItems: an associative array indicating the the failed raw row under the 'raw_row' key. This will be an empty array if the line was delimited.
```

Comment:
```suggestion
   *     - failedItems: an associative array indicating the failed raw row under the 'raw_row' key. This will be an empty array if the line was delimited.
```